### DISCUSSION TITLE
How can I test a local clippy version?

### DISCUSSION THREAD

**Discussion by tisonkun** [rust-lang/rust-clippy#11294] [General]
When debugging https://github.com/rust-lang/rust-clippy/issues/7479, I clone this repo and build my own `clippy-driver` and `cargo-clippy`. After passing the `DYLD_LIBRARY_PATH` it can run with `-h` but fail with:

```
$ DYLD_LIBRARY_PATH=/Users/tison/.rustup/toolchains/nightly-2023-07-28-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib  ~/Brittani/rust-clippy/target/debug/cargo-clippy
error: process didn't exit successfully: `/Users/tison/.rustup/toolchains/nightly-2023-08-01-aarch64-apple-darwin/bin/rustc -vV` (signal: 6, SIGABRT: process abort signal)
--- stderr
dyld[68457]: missing symbol called
```

I wonder if there is a smooth way to test a local compile clippy like `cargo clippy --binpath ...` or alternative I can test out my modifications.

**Comment by flip1995** [comment]
`cargo dev lint file.rs` or `cargo dev lint /path/to/crate` to either lint a file or an entire crate with your local changes to Clippy.

When submitting a PR you'll have to submit a standalone test in `tests/ui` though. You can run only a single test with `TESTNAME=your_test_file_name cargo uitest`.
